applications/MappingApplication/custom_utilities/mapper_communicator.h

```diff
@@ -352,9 +389,149 @@ class MapperCommunicator
     {
         mpSearchStructure->Search(InitialSearchRadius,
                                   MaxSearchIterations);
-        if (mEchoLevel > 3)
+        if (mEchoLevel >= 4)
         {
-            PrintPairs();
+            // PrintPairs(); // TODO reimplement!
+        }
+    }
+
+    // CommRank is used as input bcs the MyPID function of the non-MPI MapperCommunicator is used
+    // since this function is called before the MapperMPICommunicato is initialized
+    void CheckAndValidateJson(const int CommRank)
+    {
+        // Check if there is a valid input for the search parameters
+        bool compute_search_radius = true;
+        if (mJsonParameters.Has("search_radius"))
```

Comment:
just a curiosity. why didn't you use the Validate method? i think that would have been more concise than this...